pages/css-grid-superposition--qcj5jo/css-grid-superposition--qcj5jo.md

```diff
@@ -0,0 +1,279 @@
+
+The available space was too small to add a title at the top or bottom of the chart so to help our users figure this out, I added an information button ℹ️ in the top right corner.
+When they click on it, a short text is displayed in place of the chart like this:
+
+![UI Component displaying some help](./cc-tile-status-codes-help.png)
+
+When they click on the cross icon, we hide the short text and display the chart again.
+Implementing this toggle behaviour was not that hard.
+Let's take a simplified version of the template:
+
+```html
+<div class="chart-component">
+  <div class="title">
+    HTTP response codes <button>toggle chart/info</button>
+  </div>
+  <div class="chart">
+    <!-- chart here -->
+  </div>
+  <div class="info">
+    <!-- short text here -->
+  </div>
+</div>
+```
+
+With this HTML, we could simply toggle the component state between `.chart` and `.info` when the button is clicked and hide the appropriate panel with `display: none`.
+
+## Layout shift problems
+
+This approach has one big problem, it is subject to **layout shift**.
+
+> What is that?
+
+When I work on such components, I tend to follow what I consider a good practice: don't set a fixed height on the component, let it adapt to its own content.
+This is even more important in our situation because we have no way to know how much space we need to display the short text (without scrollbars).
+The minimum required height will depend on:
+
+* The unknown width of the component, which depends on usage context
+* The length of the short text, which depends on the translation (english or french)
+
+The result is, when the short text is displayed, the component height is often a bit bigger than when the chart was displayed:
+
+![UI Component displaying some help](./heigh-change-en.png)
+
+The situation is worse in french because the text is a bit longer:
+
+![UI Component displaying some help](./heigh-change-fr.png)
+
+Depending on the situation, this innocent looking height change can have big impacts on the whole layout and the scroll position.
+This kind of details can drive your users crazy:
+
+* Best case scenario, other parts of the layout slightly move.
+* Worst case scenario: the button under the mouse cursor moves and something else is placed under.
+
+In the animation below, you can see what happens when we start from the "info" state and click on the "Show the graph" button.
+
+![UI Component displaying some help](./layout-shift-animation.gif)
+
+Look carefully at the mouse cursor.
+It does not move but a simple click on the button triggers a chain reaction:
+
+* The height of the component is reduced to display the chart.
+* The height of the whole page is reduced.
+* The scrollbar is "disabled" (not necessary anymore).
+* The whole page is scrolled back to top and we can see the header again.
+
+As you can see, when the chart is displayed, the mouse cursor is now above some text.
+
```

Comment:
I would add something like "Now, how can we prevent this layout shift from happening?"